### PR TITLE
add git tags and docker labels

### PR DESCRIPTION
**Pull request by alberhander**
- Adding docker labels for images (following [opencontainers standard](https://github.com/opencontainers/image-spec/blob/master/annotations.md)):
  - commit_sha
  - creation_date
- Pushing git tags when present
- Specify `--build-arg COMPILE_ASSETS` in all workflows 
- Fix substitutions issue while applying labels (TLDR; you need to use double `$$` when doing var substitution in bash, if `substitutions` is enabled on the build): 
Following this article: https://medium.com/@davidstanke/mastering-google-cloud-build-config-syntax-8c3024607daf and thanks to @ManuelLR too!
```
Step 48/49 : LABEL org.opencontainers.image.created=2021-02-08 11:03:27+00:00
Step #7:  ---> Running in 9f07bb23cc88
Step #7: Removing intermediate container 9f07bb23cc88
Step #7:  ---> e9be851aeb46
Step #7: Step 49/49 : LABEL org.opencontainers.image.revision=2734f7869364cd93a92555672b95a818b848b84f
Step #7:  ---> Running in c3fac785f123
```

### CODE REVIEW

**Comment by ManuelLR** [comment]
The use of that standard for labels seems to be fine. :ok_hand: 

A question, Why you delete `COMPILE_ASSETS` build-arg ?

**Comment by alberhander** [comment]
> The use of that standard for labels seems to be fine. 
> 
> A question, Why you delete `COMPILE_ASSETS` build-arg ?

because I'm setting it to `true` by default in the Dockerfile.[ PR](https://github.com/CartoDB/cartodb-private/pull/17)

**Comment by alberhander** [comment]
@ilbambino please let me know if you think the labels we added are enough.
@ManuelLR I'll wait for you to merge this one, as it bumps `private` and there are changes you introduced in cartodb-private.

**Review by ilbambino — APPROVED** [review]
With those two it's enough to pinpoint a commit in git, so 👍 
But I wonder why did you add the labels only to `master` and not to all images? Wouldn't it make sense? For homogeneity and just in case at some point we need to make a hotfix in a branch for a past release?

File: script/ci/cloudbuild-new-pr-pg11.yaml [hunk 1]
```diff
@@ -81,11 +81,11 @@ steps:
           docker pull gcr.io/cartodb-on-gcp-main-artifacts/builder:latest
           docker tag gcr.io/cartodb-on-gcp-main-artifacts/builder:latest gcr.io/cartodb-on-gcp-main-artifacts/builder:${_BRANCH_TAG}
         fi
-      docker build --build-arg BUNDLE_JOBS=16 -t gcr.io/cartodb-on-gcp-main-artifacts/builder:current -t gcr.io/cartodb-on-gcp-main-artifacts/builder:${_BRANCH_TAG} -t gcr.io/cartodb-on-gcp-main-artifacts/builder:${SHORT_SHA} -t gcr.io/cartodb-on-gcp-main-artifacts/builder:${_BRANCH_TAG}--${SHORT_SHA} --cache-from gcr.io/cartodb-on-gcp-main-artifacts/builder:${_BRANCH_TAG} .
+      docker build --build-arg COMPILE_ASSETS=false --build-arg BUNDLE_JOBS=16 -t gcr.io/cartodb-on-gcp-main-artifacts/builder:current -t gcr.io/cartodb-on-gcp-main-artifacts/builder:${_BRANCH_TAG} -t gcr.io/cartodb-on-gcp-main-artifacts/builder:${SHORT_SHA} -t gcr.io/cartodb-on-gcp-main-artifacts/builder:${_BRANCH_TAG}--${SHORT_SHA} --cache-from gcr.io/cartodb-on-gcp-main-artifacts/builder:${_BRANCH_TAG} .
 
 # Run tests
 - name: 'docker/compose:1.22.0'
-  args: ['-f', 'docker-compose-pg11.yml', 'up', '--build', '-d']
+  args: ['-f', 'docker-compose-pg11.yml', 'up', 'q--build', '-d']
```
Comment (ManuelLR) [inline script/ci/cloudbuild-new-pr-pg11.yaml:88]:
```suggestion
  args: ['-f', 'docker-compose-pg11.yml', 'up', '--build', '-d']
```

Comment (alberhander) [reply script/ci/cloudbuild-new-pr-pg11.yaml:88]: thanks! fixing it

**Pull request merged**